content/browser/secure_embed_connector_impl.cc

```diff
@@ -649,6 +671,17 @@ void SecureEmbedConnectorImpl::OnRenderViewReady() {
   UpdateViewForCurrentRenderFrameHost();
 }
 
+void SecureEmbedConnectorImpl::EmbedderVisibilityChanged(
+    content::Visibility visibility) {
+  // When the embedder visibility changes, we may need to update the embedded
+  // visibility as well. Changes in the visibility at the embedder WebContents
+  // level don't automatically show up as a visibility change at the embed/
+  // plugin level.
+  if (delegate_) {
+    delegate_->EmbedderVisibilityChanged(visibility);
```

Comment:
Hmm. This appears to end up just calling SetVisibility again (via SecureEmbedHost::EmbedderVisibilityChanged), which feels wrong to me?
Like you can have embedder visibility hide things, then  visibility: on the plugin change to visible, and we will end up calling guest_web_contents_->WasShown() on line 547? It feels like these should be two bits of state that are combined

Reply:
+1. The result visibility should be the AND'ed result of two sources of visibilities:`plugin_element_visibility_` and `embedder_web_contents_visibility_`. Can we store the 2 states in `SecureEmbedHost`?